plugins/woocommerce/tests/php/src/Internal/Admin/Settings/PaymentProviders/WooPayments/WooPaymentsServiceTest.php

```diff
@@ -0,0 +1,4903 @@
+		}
+
+		$this->assertFalse( $this->sut->is_valid_onboarding_step_id( 'invalid_step' ) );
+	}
+
+	/**
+	 * Test get_onboarding_step_status.
+	 *
+	 * @dataProvider provider_get_onboarding_step_status
+	 *
+	 * @param string $step_id              The step ID.
+	 * @param string $expected_status      The expected status.
+	 * @param array  $step_stored_statuses The stored statuses for the step.
+	 * @param array  $wpcom_connection     The WPCOM connection state.
+	 * @param array  $account_state        The account state.
+	 *
+	 * @return void
+	 * @throws \Exception On invalid mocking.
+	 */
+	public function test_get_onboarding_step_status( string $step_id, string $expected_status, array $step_stored_statuses, array $wpcom_connection, array $account_state ) {
+		$location = 'US';
+
+		// Arrange.
+		$stored_profile = array(
+			'onboarding' => array(
+				$location => array(
+					'steps' => array(
+						$step_id => array(
+							'statuses' => $step_stored_statuses,
+						),
+					),
+				),
+			),
+		);
+
+		$this->mockable_proxy->register_function_mocks(
+			array(
+				'class_exists' => function ( $class_to_check ) {
+					if ( '\WC_Payments' === $class_to_check ) {
+						return true;
+					}
+
+					return false;
+				},
+				'get_option'   => function ( $option_name, $default_value = null ) use ( $stored_profile ) {
+					if ( WooPaymentsService::NOX_PROFILE_OPTION_KEY === $option_name ) {
+						return $stored_profile;
+					}
+
+					return $default_value;
+				},
+			)
+		);
+
+		// Arrange the WPCOM connection.
+		$this->mock_wpcom_connection_manager
+			->expects( $this->any() )
+			->method( 'is_connected' )
+			->willReturn( $wpcom_connection['is_store_connected'] );
+		$this->mock_wpcom_connection_manager
+			->expects( $this->any() )
+			->method( 'has_connected_owner' )
+			->willReturn( $wpcom_connection['has_connected_owner'] );
+
+		// Arrange the account.
+		$this->mock_provider
+			->expects( $this->any() )
+			->method( 'is_account_connected' )
+			->willReturn( $account_state['has_account'] );
+
+		$this->mock_account_service
+			->expects( $this->any() )
+			->method( 'is_stripe_account_valid' )
+			->willReturn( $account_state['has_valid_account'] );
+		$this->mock_account_service
+			->expects( $this->any() )
+			->method( 'get_account_status_data' )
+			->willReturn(
+				array(
+					'status'           => 'complete',
+					// These two are mirror images of each other.
+					'testDrive'        => $account_state['test_account'] ?? false,
+					'isLive'           => ! $account_state['test_account'] ?? true,
+					'paymentsEnabled'  => true,
+					'detailsSubmitted' => true,
+				)
+			);
+
+		// Act.
+		$result = $this->sut->get_onboarding_step_status( $step_id, $location );
+
+		// Assert.
+		$this->assertSame( $expected_status, $result );
+	}
+
+	/**
+	 * Data provider for test_get_onboarding_step_status.
+	 *
+	 * @return array[]
+	 */
+	public function provider_get_onboarding_step_status(): array {
```

Comment:
this test seem to cover smaller part of the test above, but I guess we should keep it for better safety

Reply:
Do you mean the clean slate case or all the `test_get_onboarding_step_status`?

Reply:
I meant `test_get_onboarding_details_steps` and `provider_get_onboarding_details_steps`

Reply:
yes, that's fair, that's why I said we should keep it!